### PR TITLE
Migrate the frontend to use the new yank API

### PR DESCRIPTION
**Pull request by Rustin170506**
Moved to the new API before adding frontend support for the yank message.

Tested locally:

https://github.com/user-attachments/assets/bf3ce6c2-b6b5-499c-b878-5180e710fd7d

### CODE REVIEW

**Review by Rustin170506 — COMMENTED** [review]
🔢 Self-check (PR reviewed by myself and ready for feedback.)

File: mirage/route-handlers/crates.js [hunk 1]
```diff
@@ -264,6 +264,21 @@ export function register(server) {
...
+    if (!crate) {
+      return notFound();
+    }
+
+    const version = schema.versions.findBy({ crateId: crate.id, num: versionNum });
+    if (!version) {
+      return notFound();
+    }
+
+    return {};
```
Comment (Turbo87) [inline mirage/route-handlers/crates.js:279]: we should update the `yank` status and add the `yank_message` field here and then return the serialized version. note that there is also a test suite for the mirage route handlers to ensure that it roughly matches the real API that we are mocking here. if we implement the frontend code against an API that does not match the real API we're in for trouble 😅

Comment (Rustin170506) [reply mirage/route-handlers/crates.js:279]: I followed the previous mock API 😅
```js
  server.delete('/api/v1/crates/:name/:version/yank', (schema, request) => {
    const { name, version: versionNum } = request.params;
    const crate = schema.crates.findBy({ name });
    if (!crate) {
      return notFound();
    }

    const version = schema.versions.findBy({ crateId: crate.id, num: versionNum });
    if (!version) {
      return notFound();
    }

    return {};
  });

  server.put('/api/v1/crates/:name/:version/unyank', (schema, request) => {
    const { name, version: versionNum } = request.params;
    const crate = schema.crates.findBy({ name });
    if (!crate) {
      return notFound();
    }

    const version = schema.versions.findBy({ crateId: crate.id, num: versionNum });
    if (!version) {
      return notFound();
    }

    return {};
  });
```

Do we need to update the old one as well? Or just delete it?

Comment (Turbo87) [reply mirage/route-handlers/crates.js:279]: ouch, wow, yeah, those are definitely wrong 🙈

Comment (Rustin170506) [reply mirage/route-handlers/crates.js:279]: Alright, I’ll update it as well. Initially, I thought we were focusing only on UI testing(we only checked the button name), so I didn’t consider the actual version status. I’ll make sure to update the version status correctly.

Comment (Turbo87) [reply mirage/route-handlers/crates.js:279]: > I thought we were focusing only on UI testing(we only checked the button name), so I didn’t consider the actual version status

we have the "crates can be yanked by owner" test, but that apparently only asserts local state, which meant that the bug in the mirage implementation was hidden. and apparently we currently don't have any tests for the yank/unyank endpoints in mirage 😢

**Review by Rustin170506 — COMMENTED** [review]
🔢 Self-check (PR reviewed by myself and ready for feedback.)

**Pull request merged**